### ISSUE TITLE
Some way to prevent dependency installation via apt/yum/etc.?

### ISSUE THREAD

**Issue by oschulz** [JuliaLang/BinDeps.jl#248]
It would be great if there was some way (environment variable?) for the user to force BinDeps *not* to use `sudo apt-get ...`, `sudo yum ...`, but to always download and compile dependencies instead (if they're not already installed):

* The user may not have sudo privileges (e.g. on a typical multi-user system, compute cluster, etc.). Currently, this forces him to download, build and install the dependency by hand when BinDeps insists on doing sudo.

* Some of the deps to be installed (e.g. libzmq3-dev for ZMQ) may cause conflicts with another frequently used packages (e.g. libzmq-dev, which is zmq-2.2.0 on Ubuntu Xenial). I have this exact problem right now - I can't ask our sysadmins to install libzmq3-dev on our systems, because that would prevent installation of libzmq-dev.

* The user may want to ensure that the default version of the dependency (usually a recent one) is installed, instead of the one available from the system package manager. The system version may be too old for his purpose, or the user may want to ensure a consistent software set across different systems.

Sorry if there's already a workaround for this - I didn't find any, so far.

**Comment by nalimilan** [comment]
Makes sense. See also https://github.com/JuliaLang/BinDeps.jl/issues/210.

Though regarding the example you give, you shouldn't have to install -dev packages. This is a bug in ZMQ.jl, which should load versioned .so files instead.

**Comment by oschulz** [comment]
Thanks, good point, @nalimilan. I thought ZMQ really needed the -dev package for some reason, but it seems to be perfectly happy with just "libzmq3" installed. I've created a PR (JuliaInterop/ZMQ.jl#127).

**Comment by azric** [comment]
I'm in a similar situation to oschulz as a cluster manager; users are not permitted to run sudo, period.  The problem is a user doing
   Pkg.add("Gtk")
sets off alarms about unauthorized use of sudo, which confuses the user, and also results in their account being disabled.  In today's unfriendly world, sudo should only be attempted by authorized system owners/administrators.  It would be nice if Julia could be told by some site specific means (config file, environment variable setting, build option...) not to attempt sudo.

**Comment by oschulz** [comment]
I agree with @azric  - in bigger,  more "formal" computing environments, things like automatic sudo can be very problematic. I try to prevent this by providing software containers with all dependencies included to our users, but it's hard to predict what an additional package will suddenly try to install ...
